core/writer/kafka_handler.go

```diff
@@ -110,129 +110,224 @@ func (k *KafkaDataHandler) KafkaOp(ctx context.Context, database string, f func(
-		val := fmt.Sprintf("flush collections %v", param.CollectionNames)
-		return k.sendMessage(p, []byte(val), dc)
+		return k.sendMessage(p, param, flushInfo, dc)
 	})
 }
 
-func (k *KafkaDataHandler) sendMessage(p *kafka.Producer, val []byte, dc chan kafka.Event) error {
+func (k *KafkaDataHandler) ReplicateMessage(ctx context.Context, param *api.ReplicateMessageParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, replicateMessageInfo, dc)
+	})
+}
+
+// rbac messages
+func (k *KafkaDataHandler) CreateUser(ctx context.Context, param *api.CreateUserParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, createUserInfo, dc)
+	})
+}
+
+func (k *KafkaDataHandler) DeleteUser(ctx context.Context, param *api.DeleteUserParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, deleteUserInfo, dc)
+	})
+}
+
+func (k *KafkaDataHandler) UpdateUser(ctx context.Context, param *api.UpdateUserParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, updateUserInfo, dc)
+	})
+}
+
+func (k *KafkaDataHandler) CreateRole(ctx context.Context, param *api.CreateRoleParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, createRoleInfo, dc)
+	})
+}
+
+func (k *KafkaDataHandler) DropRole(ctx context.Context, param *api.DropRoleParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		return k.sendMessage(p, param, dropRoleInfo, dc)
+	})
+}
+
+func (k *KafkaDataHandler) OperateUserRole(ctx context.Context, param *api.OperateUserRoleParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		var info msgType
+		switch param.Type {
+		case milvuspb.OperateUserRoleType_AddUserToRole:
+			info = addUserToRoleInfo
+		case milvuspb.OperateUserRoleType_RemoveUserFromRole:
+			info = removeUserFromRoleInfo
+		default:
+			log.Warn("unknown operate user role type", zap.String("type", param.Type.String()))
+			return nil
+		}
+		return k.sendMessage(p, param, info, dc)
+	})
+}
+
+func (k *KafkaDataHandler) OperatePrivilege(ctx context.Context, param *api.OperatePrivilegeParam) error {
+	return k.KafkaOp(ctx, "", func(p *kafka.Producer, dc chan kafka.Event) error {
+		var info msgType
+		switch param.Type {
+		case milvuspb.OperatePrivilegeType_Grant:
+			info = grantPrivilegeInfo
+		case milvuspb.OperatePrivilegeType_Revoke:
+			info = revokePrivilegeInfo
+		default:
+			log.Warn("unknown operate privilege type", zap.String("type", param.Type.String()))
+			return nil
+		}
+		return k.sendMessage(p, param, info, dc)
+	})
+}
+
+func (k *KafkaDataHandler) DescribeCollection(ctx context.Context, param *api.DescribeCollectionParam) error {
```

Comment:
why the describe ops send the message to the mq? include: DescribeCollection/DescribeDatabase/DescribePartition